fido/src/main/java/com/yubico/yubikit/fido/webauthn/AuthenticatorAttestationResponse.java

```diff
@@ -127,30 +128,88 @@ public byte[] getAttestationObject() {
+                }
+                map.put(PUBLIC_KEY_ALGORITHM, publicKeyAlgorithm);
+                map.put(ATTESTATION_OBJECT, Base64.encode(attestationObject));
+                break;
+            }
+            case CBOR: {
+                map.put(CLIENT_DATA_JSON, getClientDataJson());
+                map.put(AUTHENTICATOR_DATA, authenticatorData.getBytes());
+                map.put(TRANSPORTS, transports);
+                if (publicKey != null) {
+                    map.put(PUBLIC_KEY, publicKey);
+                }
+                map.put(PUBLIC_KEY_ALGORITHM, publicKeyAlgorithm);
+                map.put(ATTESTATION_OBJECT, attestationObject);
+                break;
+            }
         }
-        map.put(PUBLIC_KEY_ALGORITHM, publicKeyAlgorithm);
-        map.put(ATTESTATION_OBJECT, Base64.encode(attestationObject));
         return map;
     }
 
+    public Map<String, ?> toMap() {
+        return toMap(SerializationType.DEFAULT);
+    }
+
     @SuppressWarnings("unchecked")
-    public static AuthenticatorAttestationResponse fromMap(Map<String, ?> map) {
-        String publicKey = (String) map.get(PUBLIC_KEY);
+    public static AuthenticatorAttestationResponse fromMap(Map<String, ?> map, SerializationType serializationType) {
+        Object publicKey = serializationType == SerializationType.JSON
+                ? (String) map.get(PUBLIC_KEY)
+                : (byte[]) map.get(PUBLIC_KEY);
         return new AuthenticatorAttestationResponse(
-                Base64.decode(Objects.requireNonNull((String) map.get(CLIENT_DATA_JSON))),
+                serializationType == SerializationType.JSON
+                        ? Base64.decode(Objects.requireNonNull((String) map.get(CLIENT_DATA_JSON)))
+                        : Objects.requireNonNull((byte[]) map.get(CLIENT_DATA_JSON)),
                 AuthenticatorData.parseFrom(ByteBuffer.wrap(
-                        Base64.decode((String) map.get(AUTHENTICATOR_DATA)))),
+                        serializationType == SerializationType.JSON
+                                ? Base64.decode((String) map.get(AUTHENTICATOR_DATA))
+                                : (byte[]) map.get(AUTHENTICATOR_DATA))),
                 (List<String>) Objects.requireNonNull(map.get(TRANSPORTS)),
-                publicKey == null ? null : Base64.decode(publicKey),
+                publicKey == null ? null : serializationType == SerializationType.JSON
+                        ? Base64.decode((String) publicKey)
+                        : (byte[]) publicKey,
                 (Integer) map.get(PUBLIC_KEY_ALGORITHM),
-                Base64.decode(Objects.requireNonNull((String) map.get(ATTESTATION_OBJECT)))
+                serializationType == SerializationType.JSON
+                        ? Base64.decode(Objects.requireNonNull((String) map.get(ATTESTATION_OBJECT)))
+                        : Objects.requireNonNull((byte[]) map.get(ATTESTATION_OBJECT))
         );
```

Comment:
In a lot of the fromMap implementations we use multiple ternary operators, but for the toMap we use switch statements. I would prefer if we consistently use switches, as the condition is then not repeated for each argument, and it makes it easier to add additional SerializationTypes in the future. The logic is a bit more explicit that way as well: JSON or CBOR, instead of JSON or NOT JSON.

Reply:
fixed